clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -3871,6 +3871,24 @@ void CodeGenModule::generateIntelFPGAAnnotation(
       break;
     }
     Out << '}';
+    if (const auto *DD = dyn_cast<DeclaratorDecl>(D)) {
+      Out << "{sizeinfo:";
+      // D can't be of type FunctionDecl (as no memory attribute can be applied
+      // to a function)
+      QualType ElementTy = DD->getType();
+      QualType TmpTy = ElementTy->isArrayType()
+                           ? getContext().getBaseElementType(ElementTy)
+                           : ElementTy;
+      Out << getContext().getTypeSizeInChars(TmpTy).getQuantity();
+      // Add to Out the dimenstion of the array.
```

Comment:
Minor grammar-nazism:
```suggestion
      // Add the dimenstion of the array to Out.
```